### PR TITLE
Separate tar archive and tar flags

### PR DESCRIPTION
**Pull request by deantvv**
It used to have one flag (-r) to enable both tar archive and tar.
Now it has two flags [ -r: for tar, -g: for tar archive].

Related to #451

### CODE REVIEW

**Comment by deantvv** [comment]
@svenstaro Could you take a look at this? Thanks in advance.

**Comment by svenstaro** [comment]
Will do, sorry, pretty busy these days.

**Review by svenstaro — CHANGES_REQUESTED** [review]
I really like this and would basically accept it as is code-wise but I'd really like to distinguish the terminology here: Let's call uncompressed tarballs `uncompressed tar archives` and compressed ones `gz-compressed tar archives` for the sake of being able to add on more compression methods later on without it being to confusing sounding.

Keep in mind: tar are _always_ archives, whether they are compressed or not.

File: src/archive.rs
Comment (svenstaro) [inline src/archive.rs:58]: Let's go with `tar_enabled` and `tar_compressed_enabled`.

File: src/args.rs
Comment (svenstaro) [inline src/args.rs:102]: `/// Enable uncompressed tar archive generation`

File: src/args.rs
Comment (svenstaro) [inline src/args.rs:106]: `/// Enable gz-compressed tar archive generation`

**Review by svenstaro — CHANGES_REQUESTED** [review]
As per my other longer comment, I gave you some in-code hints for what I mean. Could you change that code and the rest of your changes to conform to those suggestions?

File: tests/archive.rs
Comment (svenstaro) [inline tests/archive.rs:55]: Good stuff, I like tests! :D

File: src/archive.rs
Comment (deantvv) [reply src/archive.rs]: Yes, it is now changed.

File: tests/archive.rs
Comment (deantvv) [reply tests/archive.rs:55]: Me too!

File: src/archive.rs
Comment (svenstaro) [inline src/archive.rs:58]: Come to think of it, it probably wasn't proper of me to go with `CompressionMethod` here as it should really have been `ArchiveMethod` from the beginning. Oh well, might change this after merging your PR. :)

**Comment by deantvv** [comment]
> I really like this and would basically accept it as is code-wise but I'd really like to distinguish the terminology here: Let's call uncompressed tarballs `uncompressed tar archives` and compressed ones `gz-compressed tar archives` for the sake of being able to add on more compression methods later on without it being to confusing sounding.
> 
> Keep in mind: tar are _always_ archives, whether they are compressed or not.

Thanks for the suggestions and fixing the wrong terminology. It is now fixed in c9506c7.

**Comment by svenstaro** [comment]
Cool stuff! Will merge right once CI passes.

File: tests/archive.rs
Comment (svenstaro) [inline tests/archive.rs:67]: One small thing I just found:
`// Ensure the links to the compressed tar archive exists and link to plain tar doesn't exist`

File: tests/archive.rs
Comment (svenstaro) [inline tests/archive.rs:74]: `    // Try to download, only tar.gz should work`

File: tests/archive.rs
Comment (svenstaro) [inline tests/archive.rs:55]: If you wanna do something cool here, you could use rstest for test parameterization to test `-p` and `-g` with very little test code overhead. Want to give that a try?

Comment (svenstaro) [reply tests/archive.rs:55]: Actually, on second thought, I want to more properly clean up these tests anyhow. I'll just do that after merging.

File: tests/archive.rs
Comment (deantvv) [reply tests/archive.rs:67]: I am happy to fix this. But since you already merge this, should I still change this?

File: tests/archive.rs
Comment (deantvv) [reply tests/archive.rs:55]: I can try in maybe another PR?

File: tests/archive.rs
Comment (deantvv) [reply tests/archive.rs:74]: Same as above.

File: tests/archive.rs
Comment (svenstaro) [reply tests/archive.rs:55]: Sure, if you like, that'd be much appreciated. So what I'm thinking here: Have a single parametrized test in `tests/archive.rs` that would test for all possible combinations of compression flags whether the path is reachable whether the links are being rendered. Thanks to rstest, we'd end up with a bunch of separately generated distinct tests. I think that'd be rather clean and make quite a bit of sense. Wanna give it a shot?

Take a look at: https://docs.rs/rstest/0.7.0/rstest/attr.rstest.html#test-parametrized-cases

**Pull request merged**
